src/main/java/core/basesyntax/dao/impl/CsvReadDaoImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.dao.impl;
+
+import core.basesyntax.dao.CsvDao;
+import core.basesyntax.dao.ReadDao;
+import core.basesyntax.model.fruit.Record;
+import core.basesyntax.service.parser.ReadParser;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+public class CsvReadDaoImpl extends CsvDao implements ReadDao {
+    private final ReadParser parser;
+
+    public CsvReadDaoImpl(String inputPath, ReadParser parser) {
+        super(inputPath);
+        this.parser = parser;
+    }
+
+    @Override
+    public List<Record> read() {
```

Comment:
Lets this method returns `List<String>`, so this class can be not dependent from `ReadParser`.
This make class to have  more compliance to **Single Responsibility** principle, and can be more flexible for the future updates.